### PR TITLE
feat: Refactor journal tool to record with immutable events.

### PR DESCRIPTION
**Pull request by msumari**
## Summary
Refactored the journal tool to use the shared record_event function and EventStatus constants, implementing an immutable event-
based approach .

### Journal Tool (src/tools/journal.py)
• **Replaced DynamoDB operations** with record_event calls
• **Removed session management** - now only handles task/phase tracking
• **Added phase names to event statuses** for easier identification (e.g., TASK_STARTED_DISCOVERY)
• **Added input validation** for action and status parameters
• **Simplified to 2 actions**: start_task and complete_task

### Event Status Constants (src/shared/event_statuses.py)
• **Added task tracking constants**: TASK_STARTED, TASK_COMPLETED, TASK_FAILED, TASK_CANCELLED, TASK_SKIPPED

### Agent Prompt (src/agents/prompt.md)
• **Removed session management instructions** (update_session, complete_session)
• **Updated workflow** to focus only on task/phase tracking
• **Simplified journaling pattern** to match new tool capabilities

### Test Infrastructure
• **Fixed @tool decorator mocking** in conftest.py for consistent test behavior
• **Added comprehensive edge case tests**: invalid actions, invalid status, missing env vars, special characters
• **Updated existing tests** to use record_event mocking instead of DynamoDB

## Testing
• ✅ All tests pass
• ✅ Backward compatibility maintained for existing functionality

### CODE REVIEW

File: src/agents/prompt.md [hunk 1]
```diff
@@ -36,40 +36,28 @@ Use calculator for all calculations. Never do math mentally.
...
 **What You Can Do:**
 
-- Start a new session to begin tracking your work
 - Mark when you begin each major phase of work
 - Record when phases complete successfully or encounter issues
-- Finish the entire session when all work is done
 
 **How to Mark Completion Status:**
 
 - For tasks: COMPLETED, FAILED, CANCELLED, or SKIPPED
```
Comment (rezabekf) [inline src/agents/prompt.md:48]: Nit: I dont think we used those anymore

File: src/agents/prompt.md [hunk 2]
```diff
@@ -36,40 +36,28 @@ Use calculator for all calculations. Never do math mentally.
...
 
 **Your Workflow Pattern:**
 
-1. **Before you start any work:**
-   - Begin tracking your session with action "update_session"
-   - If this fails, note the error in "Gaps & Limitations" but continue your work
-
-2. **When starting each major phase:**
+1. **When starting each major phase:**
    - Mark the phase start with action "start_task" and provide the phase_name
```
Comment (rezabekf) [inline src/agents/prompt.md:53]: nit: the fn(0 is called `_start_task` probably doesnt matter.

File: src/agents/prompt.md [hunk 3]
```diff
@@ -36,40 +36,28 @@ Use calculator for all calculations. Never do math mentally.
...
-   - If this fails, note the error but continue with your actual work
+   - If this fails, note the error in "Gaps & Limitations" but continue with your actual work
 
-3. **When finishing each phase:**
+2. **When finishing each phase:**
    - Mark successful completion with action "complete_task", the phase_name, and status "COMPLETED"
    - For failures, use action "complete_task", the phase_name, status "FAILED", and include an error_message
```
Comment (rezabekf) [inline src/agents/prompt.md:58]: nit: Do we still use COMPELTED/FAILED somewhere?

File: src/tools/journal.py [hunk 4]
```diff
@@ -287,119 +108,40 @@ def _complete_task(
...
-            "phase_name": phase_name,
-            "status": status.value,
-            "duration_seconds": duration_seconds,
-            "timestamp": end_time,
+        # Map string status to EventStatus constants
+        status_map = {
+            "COMPLETED": EventStatus.TASK_COMPLETED,
+            "FAILED": EventStatus.TASK_FAILED,
+            "CANCELLED": EventStatus.TASK_CANCELLED,
+            "SKIPPED": EventStatus.TASK_SKIPPED,
```
Comment (rezabekf) [inline src/tools/journal.py:113]: nit: shall we match key values same, e.g `"TASK_COMPLETED": EventStatus.TASK_COMPLETED,`

File: src/tools/journal.py [hunk 5]
```diff
@@ -72,125 +49,6 @@ def _create_timestamp_and_ttl() -> tuple[str, int]:
     return timestamp, ttl
```
Comment (rezabekf) [inline src/tools/journal.py]: do we need `_create_timestamp_and_ttl()` fn? this is handled by recorder, i think all we need is set at the top `ttl_days = int(os.environ.get("TTL_DAYS", "90"))`
and then in recorder we just set 
```record_event(
            session_id=session_id,
            status=event_status,
            table_name=journal_table_name,
            ttl_days=ttl_days,
            error_message=f"{error_code} - {error_message}",
            region_name=aws_region,
        )
```

File: src/tools/journal.py [hunk 6]
```diff
@@ -209,28 +67,17 @@ def _start_task(phase_name: str, tool_context: ToolContext) -> Dict[str, Any]:
...
-                start_time=timestamp,
-            )
-            table.put_item(Item=item)
-        except ClientError as e:
-            return _handle_dynamodb_error("create task", e, {"session_id": session_id, "phase_name": phase_name})
+        record_event(
+            session_id=session_id,
+            status=event_status,
+            table_name=table_name,
+            ttl_days=ttl // (24 * 60 * 60),
```
Comment (rezabekf) [inline src/tools/journal.py]: as per coment above `ttl_days=ttl_days`

File: src/tools/journal.py [hunk 7]
```diff
@@ -209,28 +67,17 @@ def _start_task(phase_name: str, tool_context: ToolContext) -> Dict[str, Any]:
...
+            region_name=os.environ.get("AWS_REGION", "us-east-1"),
```
Comment (rezabekf) [inline src/tools/journal.py]: we can set this at the top and reuse
```
aws_region = os.environ.get("AWS_REGION", "us-east-1")

...

aws_region=aws_region
```

**Review by rezabekf — CHANGES_REQUESTED** [review]
Hi @msumari, overall, this is pretty good. I have added few nitpicks. One question, can the tool get environment variables? I have suggested changes to the way we calculate `ttl_days` if we can get env vars we could further simplify this. Idealy if we use env var, then we will not have to calculate our timestamp (recorder already does that) and we can just chage ttl in one place

File: src/tools/journal.py [hunk 8]
```diff
@@ -414,33 +156,38 @@ def journal(
...
-    Returns:
-        Dictionary with success status and operation results
+    DynamoDB journaling tool for session and task tracking.
+
+       Args:
+           action: Action to perform ( start_task,
+               complete_task)
+           phase_name: Name of the task/phase (required for start_task and
+               complete_task)
+           status: Status for completion (BUSY, COMPLETED, FAILED)
```
Comment (rezabekf) [inline src/tools/journal.py]: nit: this seems inconsistent, should be `COMPLETED, FAILED, CANCELLED, SKIPPED`

File: src/agents/prompt.md (hunk 1 above)
Comment (msumari) [reply src/agents/prompt.md:48]: The agent LLM still uses them for more easy communication then we translate them to what we store on the journal. So they are only in the prompt.

File: src/agents/prompt.md (hunk 2 above)
Comment (msumari) [reply src/agents/prompt.md:53]: no problem in this on the prompt we try to maintain english like convo without bringing much technicalities, inside the journal we have the condition that checks that action and calls appropiate function
```
if action == "start_task":
        if not phase_name:
            return _create_error_response("phase_name is required for start_task action")
        return _start_task(phase_name, tool_context)
```

File: src/agents/prompt.md (hunk 3 above)
Comment (msumari) [reply src/agents/prompt.md:58]: Same explanation as above comments

File: src/tools/journal.py (hunk 4 above)
Comment (msumari) [reply src/tools/journal.py:113]: These statuses are comming from LLM with the instruction from the prompt where we are trying to maintain natural language hence we translate them to the statuses we want to track in the Journal

File: src/tools/journal.py (hunk 7 above)
Comment (msumari) [reply src/tools/journal.py]: nice catch , changed

File: src/tools/journal.py (hunk 5 above)
Comment (msumari) [reply src/tools/journal.py]: Good point.

File: src/tools/journal.py (hunk 8 above)
Comment (msumari) [reply src/tools/journal.py]: Good catch

File: src/tools/journal.py (hunk 6 above)
Comment (msumari) [reply src/tools/journal.py]: Changed

**Review by rezabekf — APPROVED** [review]
lgmt
